jaxb/src/main/java/feign/jaxb/JAXBDecoder.java

```diff
@@ -57,11 +66,28 @@ public Object decode(Response response, Type type) throws IOException {
       throw new UnsupportedOperationException(
           "JAXB only supports decoding raw types. Found " + type);
     }
+
+
     try {
+      SAXParserFactory saxParserFactory = SAXParserFactory.newInstance();
+      saxParserFactory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+      saxParserFactory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+      saxParserFactory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false);
+      saxParserFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+
+      Source source = new SAXSource(saxParserFactory.newSAXParser().getXMLReader(), new InputSource(response.body().asInputStream()));
       Unmarshaller unmarshaller = jaxbContextFactory.createUnmarshaller((Class) type);
-      return unmarshaller.unmarshal(response.body().asInputStream());
+      return unmarshaller.unmarshal(source);
     } catch (JAXBException e) {
       throw new DecodeException(e.toString(), e);
+    } catch (ParserConfigurationException e) {
+      throw new DecodeException(e.toString(), e);
+    } catch (SAXNotRecognizedException e) {
```

Comment:
SAXNotRecognizedException and SAXNotSupportedException are subtypes of SAXException, so these catches and the imports can be removed (to help reduce clutter)